### ISSUE TITLE
README issues

### ISSUE THREAD

**Issue by lupa18** [erseco/alpine-moodle#73]
![imagen](https://github.com/user-attachments/assets/691120ee-ec66-4afd-b38b-4d3f039b50b9)

README at ends, shows this 2 lines: 
* one referencing a script, but I can't found it in latest images
* the other is void

Would be interested provide some methodology to update your image using docker and real data: moodledata, and some installed plugins. 

thanks!

**Comment by lupa18** [comment]
Thank you for your work ! 

I followed #1 link but can't find anything there.

I guess the best way to upgrade will be: 
1. Change docker-compose.yml to fit your desired version (do it one branch a time)
2.  ...  don't configure persistence on moodlehtml (comment docker-compose.yml line 56)
3. Recreate containers and restart
4. Manually download/copy custom modules when updater or notifications return a warning

**Comment by erseco** [comment]
Hi @lupa18,

Thanks for your suggestion! Your approach sounds great and aligns with a streamlined update process. Disabling persistence on `moodlehtml` and linking only the critical files like `config.php` and minimal required files is a smart way to ensure compatibility while keeping updates clean.

For `moodledata`, maintaining just the `filedir` in a persistent volume can be also a great idea since cache and other temporary directories (`cache`, `temp`, `localcache`, etc.) are automatically regenerated. This significantly reduces the storage overhead and avoids carrying over stale data during upgrades.

Additionally, including some logic in the startup scripts to ensure everything is properly updated/regenerated (e.g., clearing caches, running `upgrade.php`, and validating `moodledata` structure) would make the process more robust and reduce manual intervention.

This approach could be a solid foundation for a clean and maintainable Moodle Docker setup. Let me know if you’d like to collaborate on refining this further!

Best regards,  
Ernesto

**Comment by lupa18** [comment]
Yes, better than do like that, we may delete the volume before the upgrade, so better will be: 
1. Backup your instance
2. Change image to pull desired version, ie: v4.5.1
3. Down the containers, and delete moodlehtml volume
4. Up again, wait for migrations (config file should re create automatically based on env vars)
5. Take note of missing plugins and install them again

I think this approach is not optimal, since step 5 should be kind of automatized.

About persistence in **moodlehtml** I think would be better for quickly down/up and only delete it when upgrading

About persistence in **moodledata**, I tried to do what you said, but when down and up, then a "invalid parameter" appears on front-end and couldn't be fixed, until I put back the original non 'filedir' folders (previous deletion). So I guesss is important to have all moodledata as is.

**Comment by erseco** [comment]
Thanks for the detailed feedback and suggestions! 

1. **moodledata persistence**: You're right—maintaining the entire `moodledata` directory can be necessary in some setups to avoid issues like the “invalid parameter” error you encountered. This behavior is expected since Moodle depends on consistent data across several directories within `moodledata`. I'm okay with keeping the full `moodledata` persistence for now, as it ensures a smooth experience during upgrades and container restarts.

2. **moodlehtml persistence**: I agree with your point that managing this volume can be complex and error-prone. Instead of maintaining the volume, I think the best approach is not having it. This will make the workflow simpler:
   - Plugins could be installed using an environment variable, for example, `PLUGINS`, where you specify plugin names or URLs.
   - A banner in the admin panel could alert users that any plugins installed via the Moodle admin interface will be lost after a restart. We can recommend managing plugins through the `PLUGINS` variable instead.

3. **Automatic `config.php` regeneration**: The current logic for regenerating `config.php` from environment variables works well. During container restarts or upgrades, the container already includes the PHP files, so it will only need to update necessary components and download plugins. This approach is both efficient and maintainable.

By following this workflow, we reduce the risk of errors and simplify the upgrade process. Let me know if you want to help to implement these changes or refining the current setup further!

**Comment by lupa18** [comment]
Thank you ! 

>        * Plugins could be installed using an environment variable, for example, `PLUGINS`, where you specify plugin names or URLs.
>        * A banner in the admin panel could alert users that any plugins installed via the Moodle admin interface will be lost after a restart. We can recommend managing plugins through the `PLUGINS` variable instead.

Interesting. So is this already working? 

If not, we may add some entrypoint script to parse PLUGINS var (maybe name and version, or URL directly), and test if mod/myplugin not exists then installation tasks begin. But I also remember there are other components, like themes or admin components that must be located in other places than /mod

> By following this workflow, we reduce the risk of errors and simplify the upgrade process. Let me know if you want to help to implement these changes or refining the current setup further!

I think changes for now are few, maybe commenting out line 56 and adding to readme why we don't use persistence there and what's the recommended method to upgrade.

**Comment by erseco** [comment]
Hi @lupa18,

Just a quick update — support for installing plugins via **Moosh** was added in PR #87 
You can now use the command:

```bash
moosh plugin-install <plugin-name>
```

…to install any Moodle plugin directly from the command line. This works well with the `POST_CONFIGURE_COMMANDS` feature, so you can automate plugin installation during container setup or upgrades.

Let me know if you'd like to help test or refine this feature further.

Best,  
Ernesto
